libs/wingc/src/type_check.rs

```diff
@@ -2179,30 +2247,27 @@ impl<'a> TypeChecker<'a> {
 				Err(type_error) => self.variable_error(type_error),
 			},
 			Reference::NestedIdentifier { object, property } => {
-				// There's a special case where the object is actually a type and the property is either a static method or an enum variant.
-				// In this case the type might even be namespaced (recursive nested reference). We need to detect this and treat the entire
-				// object as a single reference to the type
-				if let Some(_type) = self.expr_maybe_type(object, env, statement_idx) {
-					// Currently we only support enum field access (no static methods)
-					let _type = match *_type {
-						Type::Enum(ref e) => {
-							if e.values.contains(property) {
-								_type
-							} else {
-								self.general_type_error(format!(
-									"Enum \"{}\" does not contain value \"{}\"",
-									_type, property.name
-								))
-							}
-						}
-						_ => self.general_type_error(format!("Type {} not valid in expression", _type)),
-					};
-					return VariableInfo {
-						_type,
-						reassignable: false,
-						// Since we only support enum variants here we assume they are phase independent, for static methods this should be fixed
-						flight: Phase::Independent,
+				// There's a special case where the object is actually a type and the property is either a static member or an enum variant.
+				// In this case the type might even be namespaced (recursive nested reference). We need to detect this and transform this
+				// reference into a type reference.
+				if let Some((_type, user_type_annotation)) = self.expr_maybe_type(object, env, statement_idx) {
+					// We can't get here twice, we can safely assume that if we're here the `object` part of the reference doesn't have and evaluated type yet.
+					assert!(object.evaluated_type.borrow().is_none());
+
+					// Create a type reference out of this nested reference and call ourselves again
+					let new_ref = Reference::TypeProperty {
+						_type: user_type_annotation,
+						property: property.clone(),
 					};
+					// Replace the reference with the new one, this is unsafe because `reference` isn't mutable and theoretically someone may
+					// hold anoter reference to it. But our AST doesn't hold up/cross references so this is safe as long as we return right.
+					let const_ptr = reference as *const Reference;
+					let mut_ptr = const_ptr as *mut Reference;
+					unsafe {
+						// We dont' use the return value but need to call replace so it'll drop the old value
+						_ = std::mem::replace(&mut *mut_ptr, new_ref);
```

Comment:
```suggestion
						let old_value = std::mem::replace(&mut *mut_ptr, new_ref);
						std::mem::drop(old_value);
```

Reply:
I think it'll drop the old value just because it goes out of scope. the `_ =` here is just to get rid of the compiler warning. So I don't think we need the explicit `drop`. I added the comment just to be explicit about what's going on here.